hydroflow/tests/networked.rs

```diff
@@ -1,86 +1,109 @@
 use std::{collections::HashSet, sync::mpsc::channel};
 
-use hydroflow::scheduled::{
-    ctx::RecvCtx, graph::Hydroflow, graph_ext::GraphExt, handoff::VecHandoff,
+use futures::{SinkExt, StreamExt};
+use hydroflow::{
+    builder::{
+        prelude::{BaseSurface, PullSurface, PushSurface},
+        surface::pull_handoff::HandoffPullSurface,
+        HydroflowBuilder,
+    },
+    scheduled::{ctx::OutputPort, graph_ext::GraphExt, handoff::VecHandoff},
 };
 use serde::{Deserialize, Serialize};
+use tokio::{
+    net::{TcpListener, TcpStream},
+    sync::mpsc::{Receiver, Sender},
+};
+use tokio_util::codec::{FramedRead, FramedWrite, LengthDelimitedCodec};
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Debug)]
 struct EchoRequest {
     return_addr: String,
     payload: String,
 }
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 struct EchoResponse {
     payload: String,
 }
 
-// This test sets up an echo server and runs a few connections to it.
-#[test]
-fn test_echo_server() {
-    let (log_message, all_received_messages) = channel();
-
+fn start_echo_server() -> u16 {
     let (server_port_send, server_port_recv) = std::sync::mpsc::channel();
 
     // First, spin up the server.
     {
         std::thread::spawn(move || {
             let rt = tokio::runtime::Runtime::new().unwrap();
             rt.block_on(async move {
-                let mut df = Hydroflow::new();
-
-                let (port, incoming_messages) = df.inbound_tcp_vertex().await;
-                let outbound_messages = df.outbound_tcp_vertex().await;
+                let mut builder = HydroflowBuilder::default();
 
+                let (port, incoming_messages): (_, OutputPort<VecHandoff<EchoRequest>>) =
+                    builder.hydroflow.inbound_tcp_vertex().await;
                 server_port_send.send(port).unwrap();
 
-                let (receiver_port, sender_port) =
-                    df.add_inout(move |_ctx, recv: &RecvCtx<VecHandoff<EchoRequest>>, send| {
-                        for msg in recv.take_inner() {
-                            send.give(Some((
-                                msg.return_addr,
+                let handler: HandoffPullSurface<VecHandoff<EchoRequest>> =
+                    builder.wrap_input(incoming_messages);
```

Comment:
In the future these two lines (40/41 and 44/45) should be their own method in `HydroflowBuilder` to keep the internals inside